rustdoc: document ESLint/TSC run via tidy in CI; add local reproduce commands

ali90h opened the pull request:
This adds a small subsection to the rustdoc GUI test suite page explaining where the frontend checks (ESLint and TypeScript) actually run in CI and how to mirror that locally. The goal is to remove guesswork for contributors who run JS/TS checks or GUI tests and aren’t sure whether compiletest or tidy is responsible.

Evidence (paths in the rust repo):
- ESLint install (tidy image): src/ci/docker/host-x86_64/tidy/Dockerfile
- Tidy extra checks run ESLint/TSC:
  - src/tools/tidy/src/extra_checks/mod.rs (calls rustdoc_js::{lint, es_check, typecheck})
  - src/tools/tidy/src/extra_checks/rustdoc_js.rs (spawns eslint and `tsc -p src/librustdoc/html/static/js/tsconfig.json`)
- CI runs `tidy` as a dedicated job, so ESLint & TypeScript checks execute in CI:
  - Workflow installs tidy: `src/ci/scripts/install-tidy.sh`
  - Matrix includes a `tidy` builder (see `src/ci/github-actions/jobs.yml`)
  - That job runs `./x.py test tidy -vv`, which calls `rustdoc_js::{lint, es_check, typecheck}`
    → invokes `eslint` and `tsc -p src/librustdoc/html/static/js/tsconfig.json`

- TypeScript project used in CI: src/librustdoc/html/static/js/tsconfig.json

Local verification (macOS):
- `./x.py test --stage 2 tests/rustdoc-js` → ✅ 63 tests OK
- `./x.py test --stage 2 tests/rustdoc-js-std` → ✅ 61 tests OK
- `./x.py test --stage 2 tests/rustdoc-gui` → ⚠️ 141 ran; 5 small layout/pixel diffs (expected on macOS; CI baseline is Linux)
- `./x.py test tidy -vv` → ✅ tidy passes
- Direct checks in `rust/src/librustdoc/html/static/js`:
  - `npm ci` → ✅
  - `npx tsc -p tsconfig.json --noEmit` → ⚠️ one env-specific type error (confirms TSC runs)
  - `npx eslint .` → ✅ runs clean

Makes CI behavior discoverable (tidy runs ESLint/TSC; compiletest runs the GUI/JS suites).
Gives exact local commands plus two common caveats (macOS pixel diffs; DOM vs Node types with local TS).

Addresses #2364.

GuillaumeGomez commented on src/rustdoc-internals/rustdoc-gui-test-suite.md:
````diff
@@ -7,6 +7,35 @@ For other rustdoc-specific test suites, see [Rustdoc test suites].
+To reproduce locally (mirrors CI behavior for rustdoc's frontend suites):
+
+```bash
+./x.py test --stage 2 tests/rustdoc-js
+./x.py test --stage 2 tests/rustdoc-js-std
+./x.py test --stage 2 tests/rustdoc-gui
````
They are not run in any of these cases as far as I know. The only way I know about is:

```suggestion
./x.py test tidy --extra-checks=[js]
```

ali90h commented:
Thanks for the feedback. I've moved the content out of the GUI test suite file and into the main rustdoc.md file instead. Also updated the command to use ./x.py test tidy --extra-checks=js and removed the manual linter instructions as suggested.

aDotInTheVoid commented:
CC @lolbinarycat

lolbinarycat commented on src/rustdoc-internals/rustdoc-gui-test-suite.md:
```diff
@@ -7,6 +7,7 @@ For other rustdoc-specific test suites, see [Rustdoc test suites].
 
 These use a NodeJS-based tool called [`browser-UI-test`] that uses [puppeteer] to run tests in a headless browser and check rendering and interactivity. For information on how to write this form of test, see [`tests/rustdoc-gui/README.md`][rustdoc-gui-readme] as well as [the description of the `.goml` format][goml-script]
 
+
```
nit: this linebreak doesn't need to be here

if its not too much trouble, i would appreciate you removing this extra linebreak then [squashing your commits](https://rustc-dev-guide.rust-lang.org/git.html#squash-your-commits) (after addressing all the other feedback) so the history for these files is a bit more clean (as is, this PR will show up as changing files it effectively doesn't)

lolbinarycat commented on src/rustdoc.md:
```diff
@@ -60,6 +60,17 @@ does is call the `main()` that's in this crate's `lib.rs`, though.)
 * Use `./x test tests/rustdoc*` to run the tests using a stage1
   rustdoc.
   * See [Rustdoc internals] for more information about tests.
+* Use `./x test tidy --extra-checks=js` to run ESLint/TSC checks on rustdoc's frontend code.
```
this command will also run `es-check`.  also note that `./x test tidy` will (by default) automatically run these checks if there are changes to javascript source files.

lolbinarycat commented on src/rustdoc.md:
```diff
@@ -60,6 +60,17 @@ does is call the `main()` that's in this crate's `lib.rs`, though.)
+
+### Frontend CI checks
+
+Rustdoc's JavaScript and TypeScript are checked by ESLint and `tsc` during CI, not by compiletest. These run as part of the tidy job:
```
```suggestion
Rustdoc's JavaScript and TypeScript are checked by `eslint`, `escheck`, and `tsc` during CI, not by compiletest. These run as part of the tidy job:
```

lolbinarycat commented on src/rustdoc.md:
```diff
@@ -60,6 +60,17 @@ does is call the `main()` that's in this crate's `lib.rs`, though.)
   * See [Rustdoc internals] for more information about tests.
+* Use `./x test tidy --extra-checks=js` to run ESLint/TSC checks on rustdoc's frontend code.
+
+### Frontend CI checks
```
```suggestion
### JavaScript CI checks
```

I think this would be a bit more clear and precise.

ali90h commented:
Thanks, @lolbinarycat  — I updated the patch per your notes

jieyouxu commented:
r? rustdoc

GuillaumeGomez commented:
Thanks!
